dask/dataframe/tests/test_dataframe.py

```diff
@@ -402,6 +423,58 @@ def test_reductions():
         assert eq(dds.mean(), pds.mean())
         assert eq(dds.nunique(), pds.nunique())
 
+def test_reductions_non_numeric_dtypes():
+    # test non-numric blocks
+
+    def check_raises(d, p, func):
+        assert raises(TypeError, lambda: getattr(d, func)().compute())
+        assert raises(TypeError, lambda: getattr(p, func)())
+
+    pds = pd.Series(['a', 'b', 'c', 'd', 'e'])
+    dds = dd.from_pandas(pds, 2)
+    assert eq(dds.sum(), pds.sum())
+    assert eq(dds.min(), pds.min())
+    assert eq(dds.max(), pds.max())
+    assert eq(dds.count(), pds.count())
+    check_raises(dds, pds, 'std')
+    check_raises(dds, pds, 'var')
+    check_raises(dds, pds, 'mean')
+    assert eq(dds.nunique(), pds.nunique())
+
+    for pds in [pd.Series(pd.Categorical([1, 2, 3, 4, 5], ordered=True)),
+                pd.Series(pd.Categorical(list('abcde'), ordered=True)),
+                pd.Series(pd.date_range('2011-01-01', freq='D', periods=5))]:
+        dds = dd.from_pandas(pds, 2)
+
+        check_raises(dds, pds, 'sum')
+        assert eq(dds.min(), pds.min())
+        assert eq(dds.max(), pds.max())
+        assert eq(dds.count(), pds.count())
+        check_raises(dds, pds, 'std')
+        check_raises(dds, pds, 'var')
+        check_raises(dds, pds, 'mean')
+        assert eq(dds.nunique(), pds.nunique())
+
+
+    pds= pd.Series(pd.timedelta_range('1 days', freq='D', periods=5))
+    dds = dd.from_pandas(pds, 2)
+    assert eq(dds.sum(), pds.sum())
+    assert eq(dds.min(), pds.min())
+    assert eq(dds.max(), pds.max())
+    assert eq(dds.count(), pds.count())
+
+    # ToDo: pandas supports timedelta std, otherwise dask raises:
+    # incompatible type for a datetime/timedelta operation [__pow__]
+    # assert eq(dds.std(), pds.std())
```

Comment:
For numerical stability, ``pandas`` is also changing to use ``two-pass`` algorithm.

- https://github.com/pydata/pandas/pull/10679

And ``pandas`` can calculate timedelta ``std``, not ``var``. Calculation looks to be performed to int representation.

```
s = pd.Series(pd.timedelta_range('1 day', freq='D', periods=10))
s 
# 0    1 days
# 1    2 days
# 2    3 days
# 3    4 days
# 4    5 days
# 5    6 days
# 6    7 days
# 7    8 days
# 8    9 days
# 9   10 days
# dtype: timedelta64[ns]

s.var()
# TypeError: reduction operation 'var' not allowed for this dtype

s.std()
# Timedelta('3 days 00:39:48.990594')

# cast to int
s.astype(int).std()
# 261588990594023.31

# create Timedelta from the result
pd.Timedelta(s.astype(int).std())
# Timedelta('3 days 00:39:48.990594')
```